install_files/ansible-base/roles/grsecurity/tasks/clean_packages.yml

```diff
@@ -3,16 +3,15 @@
 # old generic kernels to avoid accidental
 # boots into a less secure environment.
 - name: Remove generic kernel packages.
-  apt:
-    name: "{{ item }}"
-    state: absent
+  command: apt-get remove -y {{ item }}
   with_items:
     - linux-signed-generic
     - linux-signed-generic-lts-utopic
     - linux-signed-image-generic
     - linux-signed-image-generic-lts-utopic
-    - 'linux-image-*generic'
-    - 'linux-headers-*'
+    - linux-image-generic-lts-xenial
+    - 'linux-image-.*generic'
+    - 'linux-headers-.*'
```

Comment:
We should inspect the output and report changed status accurately, as with the subsequent commands. Especially important given that the task hasn't been working as expected.

Reply:
The following patch will accurate report changed status:

```
diff --git a/install_files/ansible-base/roles/grsecurity/tasks/clean_packages.yml b/install_files/ansible-base/roles/grsecurity/tasks/clean_packages.yml
index d721a66a..9e73cb5f 100644
--- a/install_files/ansible-base/roles/grsecurity/tasks/clean_packages.yml
+++ b/install_files/ansible-base/roles/grsecurity/tasks/clean_packages.yml
@@ -12,12 +12,15 @@
     - linux-image-generic-lts-xenial
     - 'linux-image-.*generic'
     - 'linux-headers-.*'
+  register: apt_removed_kernels
+  changed_when: "'The following packages will be REMOVED' in apt_removed_kernels.stdout"
   tags:
     - apt
 
 - name: Get list of all installed kernels.
   shell: dpkg-query -f '${Package} ${Status}\n' -W 'linux-image*' | awk '$NF == "installed"{print $1}'
   register: apt_installed_kernels
+  changed_when: false
   tags:
     - apt
 ```